### DISCUSSION TITLE
Update to Electron 12 + Telemetry questions

### DISCUSSION THREAD

**Discussion by mauricioszabo** [pulsar-edit/.github#47] [General]
After doing https://github.com/pulsar-edit/pulsar/discussions/1, I decided to try to update to Electron 12.

The good news is that it works: it's also on the branch [build-without-scripts](https://github.com/pulsar-edit/pulsar/tree/build-without-scripts), with the same issues as I told on the last discussion.

The only issue is with Electron CrashReport. Atom uses CrashReport both on the "main process" and on the "renderer process", and using on renderer is deprecated and removed on Electron 12.

Which gives us the question: do we _keep_ CrashReport? Where it will be sent? Who will read it? What about telemetry - do we keep it, or 100% remove it?

Finally, if we decide to keep, can we create a public site where everyone can actually _see the data_ we collect, as a way to be 100% transparent? WDYT?

**Comment by confused-Techie** [comment]
My 2 cents on some questions posed:
* 'Where will it be sent?' I could absolutely add an API Endpoint on the backend to collect this data, and save it.
* 'Who will read it' A hard one, but I do like that idea that ANY data collected is publicly available, and simple to request removal.
* 'Telemetry' Its of my opinion that unless we are selling something and need product insights there is not a single reason to collect telemetry. I don't see that impacting our choices on what parts of the code base to work on, or features to implement, since likely we will work on what we want to, since this is a volunteer project. While we could collect it I don't really see a need to collect it. Beyond maybe just how many people are actually using it, as a measure of community usage, but really I don't think that would be needed, since theres the GitHub download count for releases. 

Lastly if we decide to still collect crash reports, and those are publicly available, it would allow anyone to then read this crash report, and start working on a solution. And lastly if we do I think we should have the option to deny the crash report send, per report. Since users should be able to easily change their mind on consent to crash reports, at the time of the crash.

**Reply by mauricioszabo** [reply]
Well, currently Atom collects telemetry already. Crash report is another issue - it actually is broken on Electron 12, so if we decide to not collect for now, we can delete the code and decide later what to do :)

**Reply by confused-Techie** [reply]
Yeah then personally, I would think we can hold off on the crash reporter, since theres nowhere to send it, and likely not enough eyes to actively review it. As for the telemetry, I would think thats something we can all together disable. But should likely put it to a vote.

**Reply by distalx** [reply]
I like the idea of disabling crash reports for now. and disable telemetry compliantly. VSCodium, a community-driven distribution for VSCode has done the same thing with regard to telemetry.
